### DISCUSSION TITLE
Provide URL to Specific Job Step

### DISCUSSION THREAD

**Discussion by pa3ng** [community/community#26563] [Actions]
<p>I’d like to be able to access/provide a link to a specific step in a ran job.</p>
<p>For example, if I wanted to post a comment on a PR providing a link to a specific step that has relevant output to the developer, it would be nice to access it via <code>steps.step_id.url</code>.</p>
<p>In practice, something like this:</p>
<pre><code class="lang-auto">- name: post_comment
   id: post_comment
   run: ./post_comment ${{ steps.my-output-step.url }}
</code></pre>
<p>Thanks!</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>So I have a “way” you can do this, but it is a little hacky.</p>
<p>You can actually get the link to a specific step, when looking at the build log in the UI, by clicking on the line number of that step.  For example:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/mickeygoussetorg/actions-demos/runs/2411911374?check_suite_focus=true#step:2:1" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:500/500;"><img src="https://user-images.githubusercontent.com/20031479/181098478-7a170d22-94f0-43f7-8f69-b9d47873d42f.png" class="thumbnail" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png"></div>

<h3><a href="https://github.com/mickeygoussetorg/actions-demos/runs/2411911374?check_suite_focus=true#step:2:1" target="_blank" rel="noopener">Build software better, together</a></h3>

<p><a href="https://github.com/mickeygoussetorg/actions-demos/runs/2411911374?check_suite_focus=true#step:2:1" target="_blank" rel="noopener">ygoussetorg/actions-demos/runs/2411911374</a></p>

  <p><span class="label1">GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>will take you to a run of my workflow, open the second step, and place focus on the first line.  The “2” in the URL indicates that you want to go to the second step of the workflow.  The “1” indicates go to the first line of output for that step.</p>
<p>So, if you know the order of the steps, you can make this work.  You can get the workflow run number from the github context object.  Then, you can hard-code the step number, along with “1”, and set that into the PR or issue or whatever.  Yes, if you modify the order of the steps in your job, then you have to fix this, but it would work.</p>

**Comment by pa3ng** [comment]
<p>Ah, I had no idea! It is a little hacky, but it’s good enough as far as a working solution goes :+1:</p>
<p>Appreciate it!</p>

**Comment by mickeygousset** [comment]
<p>FYI I’m including this example in a talk I’m giving in June.  :slight_smile:</p>

**Comment by pa3ng** [comment]
<p>Awesome! I’d be interested in tuning in if it’s available online anywhere :+1:</p>
